src/Evolution/Systems/Cce/Actions/InitializeKleinGordonVariables.hpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Evolution/Systems/Cce/OptionTags.hpp"
+#include "ParallelAlgorithms/Initialization/MutateAssign.hpp"
+
+namespace Cce::Actions {
+
+/*!
+ * \ingroup ActionsGroup
+ * \brief Initialize the data storage for the scalar field in the
+ * `KleinGordonCharacteristicExtract` component, which is the singleton that
+ * handles the main evolution system for Klein-Gordon CCE computations.
+ *
+ * \details Sets up the \ref DataBoxGroup to be ready to take data from the
+ * worldtube component, calculate initial data, and start the hypersurface
+ * computations.
+ *
+ * \ref DataBoxGroup changes:
+ * - Modifies: nothing
+ * - Adds:
+ *  - `Tags::Variables<metavariables::klein_gordon_boundary_communication_tags>`
+ *  - `Tags::Variables<metavariables::klein_gordon_gauge_boundary_tags>`
+ *  - `Tags::Variables<metavariables::klein_gordon_scri_tags>`
+ * - Removes: nothing
+ */
+template <typename Metavariables>
+struct InitializeKleinGordonVariables {
+  using const_global_cache_tags =
+      tmpl::list<Tags::LMax, Tags::NumberOfRadialPoints>;
+
+  using klein_gordon_boundary_communication_tags = ::Tags::Variables<
+      typename Metavariables::klein_gordon_boundary_communication_tags>;
+  using klein_gordon_gauge_boundary_tags = ::Tags::Variables<
+      typename Metavariables::klein_gordon_gauge_boundary_tags>;
+  using klein_gordon_scri_tags =
+      ::Tags::Variables<typename Metavariables::klein_gordon_scri_tags>;
+
+  using simple_tags =
+      tmpl::list<klein_gordon_boundary_communication_tags,
+                 klein_gordon_gauge_boundary_tags, klein_gordon_scri_tags>;
+
+  template <typename DbTags, typename... InboxTags, typename ArrayIndex,
+            typename ActionList, typename ParallelComponent>
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTags>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    initialize_impl(make_not_null(&box));
+    return {Parallel::AlgorithmExecution::Continue, std::nullopt};
+  }
+
+  template <typename TagList>
+  static void initialize_impl(const gsl::not_null<db::DataBox<TagList>*> box) {
```

Comment:
You don't really need an `impl` here. It's only used once, so I think this code can just go in the `apply` function.

Reply:
I was thinking about SpEC CCM, where we have to avoid all the `apply` functions. But I guess linking this KG system to SpEC won't happen in the near future. I've removed it.